src/shared/projectConfiguration.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as crypto from 'crypto';
+import { machineIdSync } from 'node-machine-id';
+import { PlatformInformation } from './platform';
+import { ITelemetryReporter, getTelemetryProps } from './telemetryReporter';
+import { DotnetInfo } from './utils/dotnetInfo';
+
+export interface ProjectConfigurationMessage {
+    ProjectId: string;
+    SessionId: string;
+    OutputKind: number;
+    ProjectCapabilities: string[];
+    TargetFrameworks: string[];
+    References: string[];
+    FileExtensions: string[];
+    FileCounts: number[];
+    SdkStyleProject: boolean;
+}
+
+export function reportProjectConfigurationEvent(
+    reporter: ITelemetryReporter,
+    projectConfig: ProjectConfigurationMessage,
+    platformInfo: PlatformInformation,
+    dotnetInfo: DotnetInfo | undefined,
+    solutionPath?: string,
+    useModernNet?: boolean
+) {
+    let solutionId = '';
```

Comment:
❗ Return early without any calculations if telemetry is disabled.

Reply:
Although that skip should be happening in the LSP server side -- we shouldn't be sending notification only to ignore them _here_.

Reply:
Resolved on the server side to not send the event at all.